File: Javascript/819. Most Common Word.js
/**
 * Link: https://leetcode.com/problems/most-common-word/
 * Author: Dung Nguyen Tien (Chris)
 * @param {string} paragraph
 * @param {string[]} banned
 * @return {string}
 */
var mostCommonWord = function(paragraph, banned) {
    var arr = paragraph.split(/[\.,?!:;`'\s]+/);
    console.log(arr);
    var dict = {};
    var occur = 0;
    var commonWord = "";
    
    for(var w of arr) {
        w = w.toLowerCase();
        if (banned.indexOf(w) == -1) {
            dict[w] = (dict[w] || 0) + 1;
            
            if (dict[w] > occur) {
                occur = dict[w];
                commonWord = w;
            }
        }    
    }
    
    return commonWord;
};

File: C#/819. Most Common Word.cs
/*
 * Link: https://leetcode.com/problems/most-common-word/submissions/
 * Author: Dung Nguyen Tien (Chris)
*/

public class Solution {
    public string MostCommonWord(string paragraph, string[] banned) {
        var arr = paragraph.ToLower()
            .Split(new char[] { '!', '?', '\'', ',', ';', '.', ' ' }, StringSplitOptions.RemoveEmptyEntries);
        
        var map = new Dictionary<string, int>();
        var set = new HashSet<string>(banned);
        int max = 0;
        string res = "";
        
        foreach(var word in arr) {
            if (!set.Contains(word)) {
                if (!map.ContainsKey(word)) map.Add(word, 0);
                
                map[word]++;
                Console.WriteLine(word + " " + map[word]);
                
                if (map[word] > max) {
                    max = map[word];
                    res = word;
                }
            }
        }
        
        return res;
    }
}